sections/sprint-3/dead-reckoning.tex

```diff
@@ -0,0 +1,84 @@
+It is often used for networking games where packages about the object's kinematic state are continuously being sent from the server to the clients.
+The kinematic state of an object includes its position, velocity, acceleration, orientation and angular velocity \autocite{DeadReckoning}.
+If the client misses a package or they are not being sent often enough to have a new update for each frame, the representation of the object will jerk across the screen instead of having smooth and consistent movement.
+That is where dead reckoning can be used to predict an objects movement to make it appear more believeable.
+For the game we're building, it is important that each player's and the ball's positions are very accurate for the game to function properly.
+\\\\
+It is not possible have a new update for each frame, zero packet losses or zero latency. 
+Therefore dead reckoning is needed to achieve a believeable representation of the ball and the players' movement \autocite{DeadReckoning}.
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.6\linewidth]{dead_reckoning/dead_reckoning_example_1.PNG}
+    \caption{Linear player movement}
+    \label{fig:dead_reckoning_example_1}
+\end{figure}
+\autoref{fig:dead_reckoning_example_1} shows an example of a player's movement in the game.
+In this example, dead reckoning would be a linear problem where the position, velocity and acceleration can be used to predict where the player will move to in the future. 
+The dead reckoned position for a specific time $Q_t$ in this example can be calculated by:
+\begin{displaymath}
+    Q_t = P'_0 + V'_0T + \frac{1}{2}A'_0T^2
+\end{displaymath}
+where $ P'_0 $ is the position, $ V'_0 $ is the velocity, $ A'_0 $ is the acceleration and $T$ is the time \autocite{DeadReckoning}.
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.6\linewidth]{dead_reckoning/dead_reckoning_example_2.PNG}
+    \caption{A new update about the player's state is received from the server}
+    \label{fig:dead_reckoning_example_2}
+\end{figure}
+In \autoref{fig:dead_reckoning_example_2} we receive a new update about the player's kinematic state. 
+This results in conflicting realities when compared to the dead reckoned position calculated from the previous example. 
+In the new update the player has turned right and a new dead reckoned position must be found, but this time it becomes a lot trickier since a believeable curve must be created from where we thought the player would be and where we estimate the player will be based on the new information. Representing the new position $P'_0$ immediately would result in the player warping across the screen which would not be ideal. Instead the player is represented at $ P_0 $ where we thought the player would be and the player should then move towards a new dead reckoned position, calculated with the information from the new update \autocite{DeadReckoning}.
```

Comment:
```suggestion
In the new update the player has turned right and a new dead reckoned position must be found, but this time it becomes a lot trickier since a believable curve must be created from where we thought the player would be and where we estimate the player will be based on the new information.
Representing the new position $P'_0$ immediately would result in the player warping across the screen which would not be ideal.
Instead, the player is represented at $ P_0 $ where the player was thought to be, and the player should then move towards a new dead reckoned position, calculated with the information from the new update \autocite{DeadReckoning}.
```